feat: zustand 초기 설정 및 초기 화면 데이터 조회 구현

lgyn10 opened the pull request:
- zustand 설정
- 모든 북마크 내의 기타 북마크 폴더만 사용 `id: '2', title: '기타 북마크'
- 새탭 페이지(초기 페이지)에 데이터 조회

<img width="1417" alt="스크린샷 2024-07-03 오후 6 30 46" src="https://github.com/lgyn10/mark-tab/assets/72643542/4e6ae918-fae0-4351-b5b3-7e08a58a4b61">

lgyn10 committed:
fix need amend

lgyn10 committed:
feat: zustand 환경 설정

lgyn10 committed:
pull from origin main

lgyn10 committed:
feat: zustand 초기 설정 및 '기타 북마크' 목록 불러오기 구현

lgyn10 committed:
feat: 초기 페이지에 북마크 데이터 조회 구현

lgyn10 commented:
@yoonhb173 
zustand 사용해서 초기 화면에 데이터를 조회하는 것까지 작업을 완료했습니다.
zustand 설정 코드와 BoardContainer, Board, List 컴포넌트 코드 위주로 검토해주실 수 있으실까요..? 
감사합니다! 🙂

yoonhb173 commented on src/Test.tsx:
```diff
@@ -12,10 +19,12 @@ const Test = () => {
-    chrome.bookmarks.getSubTree(id).then((res) => {
-      console.log(res[0]);
+  const handleGetSubTreeAPI = async (id: string) => {
+    let result;
+    await chrome.bookmarks.getSubTree(id).then((res) => {
+      result = res;
     });
```
```
const result = await chrome.bookmarks.getSubTree(id);
```

await의 경우 then 콜백으로 받는 함수를 위처럼 코드를 줄여서 받을 수 있게 하는 문법이여서, await를 쓰실꺼면 위와 같이 하는게 좋을 것 같아요!

그리고 어떠한 네트워크 요청 또는, 프로미스 객체, 함수의 경우 예상하지 못한 에러가 발생 할 수 있어서, 항상 promise는 catch 콜백이 존재합니다!

catch 콜백을 활용해 에러처리를 

```
chrome.bookmarks.getSubTree(id).then((res) => {
    result = res;
}).catch((error)=>{
    // 에러 처리
});
```

```
try { 
    const result = await chrome.bookmarks.getSubTree(id);
} catch(error) {
    console.log('error', error);
}
```

형태로 처리 해주시면 될 것 같아요!

에러 처리를 하지 않으면 사용자에게 의도하지 않은 UI나 형태를 제공하게 되고, 의도하지 않은 방향으로 사용자가 접근할 가능성이 있어서 에러가 발생했을 때의 코드도 생각해보시면 좋을 것 같습니다!

lgyn10 replied:
@yoonhb173 감사합니다! 🙌🏻

yoonhb173 commented:
```
"dependencies": {
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "axios": "^1.7.2",
    "zustand": "^4.5.4",
    "framer-motion": "^11.2.12",
    "styled-components": "^6.1.11",
    "@emotion/react": "^11.11.4",
    "@emotion/styled": "^11.11.5",
    "@chakra-ui/icons": "^2.1.1",
    "@chakra-ui/react": "^2.8.2",
    "@tanstack/react-query": "^5.49.2",
    "lodash": "^4.17.21",
  },
  "devDependencies": {
    "@types/axios": "^0.14.0",
    "@types/chrome": "^0.0.268",
    "@types/lodash": "^4.17.6",
    "@types/node": "^20.14.9",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "@types/styled-components": "^5.1.34",
    "@typescript-eslint/eslint-plugin": "^7.2.0",
    "@typescript-eslint/parser": "^7.2.0",
    "@vitejs/plugin-react": "^4.2.1",
    "chrome-types": "^0.1.290",
    "eslint": "^8.57.0",
    "eslint-plugin-react-hooks": "^4.6.0",
    "eslint-plugin-react-refresh": "^0.4.6",
    "typescript": "^5.2.2",
    "vite": "^5.2.0",
    "vite-plugin-ts": "1.3.2-1",
  }
 ```
 추가로 이렇게 package.json 설정하셔야 할 것 같습니다!
 
 dependencies안에 들어가 있는 라이브러리들은 실제 빌드시에 포함되어야 할 라이브러리가 있어야 합니다!
 
 그리고 @emotion/styled랑 styled-components는 역할이 겹치니 하나만 사용하시는게 좋을 것 같아요!

lgyn10 committed:
refactor: await 문 수정 및 try-catch문 에러 처리

lgyn10 committed:
chore: dependecies 패키지 구성 재설정

dependecies : 빌드 시에 포함되어야 할 라이브러리

lgyn10 committed:
chore: @emotion/styled 패키지 제거

 @emotion/styled랑 styled-components의 역할이 같음
styled-components를 사용함으로, @emotion/styled 패키지 제거
